test/test_automatic_segmentation.py

```diff
@@ -0,0 +1,153 @@
+        return mask, image
+
+    # create an input 2d image with three objects and stack 8 of them together
+    @classmethod
+    def _get_3d_inputs(cls, shape=(8, 256, 256)):
+        mask, image = cls._get_2d_inputs(shape[-2:])
+
+        # Create volumes by stacking the input image and respective mask.
+        volume = np.stack([image] * 8)
+        labels = np.stack([mask] * 8)
+        return labels, volume
+
+    @classmethod
+    def setUpClass(cls):
+        # Input 2d data for normal and tiled segmentation.
+        cls.mask, cls.image = cls._get_2d_inputs()
+        cls.large_mask, cls.large_image = cls._get_2d_inputs(shape=(1024, 1024))
+
+        # Input 3d data for normal and tiled segmentation.
+        cls.labels, cls.volume = cls._get_3d_inputs()
+        cls.large_labels, cls.large_volume = cls._get_3d_inputs(shape=(8, 1024, 1024))
+
+    def tearDown(self):
+        # Release all unoccupied cached memory (eg. tiling requires a lot of memory)
+        device = util.get_device(None)
+        if device == "cuda":
+            import torch.cuda
+            torch.cuda.empty_cache()
+        elif device == "mps":
+            import torch.mps
+            torch.mps.empty_cache()
+
+    def test_automatic_mask_generator_2d(self):
+        from micro_sam.automatic_segmentation import automatic_instance_segmentation
+
+        mask, image = self.mask,  self.image
+        instances = automatic_instance_segmentation(
+            input_path=image, model_type=self.model_type, ndim=2, use_amg=True
+        )
+        self.assertEqual(mask.shape, instances.shape)
+
+    def test_tiled_automatic_mask_generator_2d(self):
+        from micro_sam.automatic_segmentation import automatic_instance_segmentation
+
+        mask, image = self.large_mask,  self.large_image
+        instances = automatic_instance_segmentation(
+            input_path=image,
+            model_type=self.model_type,
+            ndim=2,
+            tile_shape=self.tile_shape,
+            halo=self.halo,
+            use_amg=True,
+        )
+        self.assertEqual(mask.shape, instances.shape)
+
+    def test_instance_segmentation_with_decoder_2d(self):
+        from micro_sam.automatic_segmentation import automatic_instance_segmentation
+
+        mask, image = self.mask,  self.image
+        instances = automatic_instance_segmentation(
+            input_path=image, model_type=self.model_type_ais, ndim=2
+        )
+        self.assertEqual(mask.shape, instances.shape)
+
+    def test_tiled_instance_segmentation_with_decoder_2d(self):
+        from micro_sam.automatic_segmentation import automatic_instance_segmentation
+
+        mask, image = self.large_mask,  self.large_image
+        instances = automatic_instance_segmentation(
+            input_path=image, model_type=self.model_type, ndim=2, tile_shape=self.tile_shape, halo=self.halo,
+        )
+        self.assertEqual(mask.shape, instances.shape)
+
+    def test_automatic_mask_generator_3d(self):
```

Comment:
If this one takes too long we can also skip it.

Reply:
Yep, this does take a while as well. I'll skip this too.